lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1052,8 +1093,18 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
     if (IsVectorCompute)
       StorageClass =
           VectorComputeUtil::getVCGlobalVarStorageClass(AddressSpace);
-    else
+    else {
+      // Lower global_device and global_host address spaces that were added in
+      // SYCL as part of SYCL_INTEL_usm_address_spaces extension to just global
+      // address space if device doesn't support SPV_INTEL_usm_storage_classes
+      // extension
+      if (!BM->isAllowedToUseExtension(
```

Comment:
I think it is better to swap order: check for address space first, and only then for extension. The reason is that extension might be disabled all the time for other users and there is no sense to check for it unless you encountered something that requires it